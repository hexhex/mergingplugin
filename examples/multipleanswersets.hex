answer(Prog1) :- &hex["

pred1(x, y, z).
pred1(xx, yy, zz).
pred2(a, b) v pred2(aa, bb).

",""](Prog1).

answersets(Prog, AS) :- answer(Prog), &answersets[Prog](AS).

pred1(AS, Arg0, Arg1, Arg2) :- answersets(Prog1, AS), &tuples[Prog1, AS](Pred, 3), &arguments[Prog1, AS, Pred](RunningNr, 0, Arg0), &arguments[Prog1, AS, Pred](RunningNr, 1, Arg1), &arguments[Prog1, AS, Pred](RunningNr, 2, Arg2).

pred2(AS, Arg0, Arg1) :- answersets(Prog1, AS), &tuples[Prog1, AS](Pred, 2), &arguments[Prog1, AS, Pred](RunningNr, 0, Arg0), &arguments[Prog1, AS, Pred](RunningNr, 1, Arg1).
