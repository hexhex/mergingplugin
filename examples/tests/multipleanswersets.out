{pred2(0,a,b), pred2(1,aa,bb), pred1(0,xx,yy,zz), pred1(0,x,y,z), pred1(1,xx,yy,zz), pred1(1,x,y,z), answersets(0,0), answersets(0,1), answer(0)}
